core/src/main/java/com/yahoo/imapnio/async/data/ExtensionListInfo.java

```diff
@@ -0,0 +1,107 @@
+package com.yahoo.imapnio.async.data;
+
+import javax.annotation.Nonnull;
+
+import com.sun.mail.iap.ParsingException;
+import com.sun.mail.imap.protocol.IMAPResponse;
+import com.sun.mail.imap.protocol.ListInfo;
+
+/**
+ * This class provides the result from LIST-EXTENDED command. It extends from ListInfo so clients can continue use the data from ListInfo. The public
+ * methods provided here allows callers to easily identify if server sends the specific interested attribute or not. The attributes supported are
+ * listed in RFC5828.
+ */
+public class ExtensionListInfo extends ListInfo {
+
+    /** Attribute name for NonExistent. */
+    private static final String NON_EXISTENT = "\\NonExistent";
+
+    /** Attribute name for Subscribed. */
+    private static final String SUBSCRIBED = "\\Subscribed";
+
+    /** Attribute name for Remote. */
+    private static final String REMOTE = "\\Remote";
+
+    /** Attribute name for HasChildren. */
+    private static final String HAS_CHILDREN = "\\HasChildren";
+
+    /** Attribute name for HasNoChildren. */
+    private static final String HAS_NO_CHILDREN = "\\HasNoChildren";
+
+    /** Flag whether NonExistent attribute present. */
+    private boolean isNonExistent = false;
+
+    /** Flag whether Subscribed attribute present. */
+    private boolean isSubscribed = false;
+
+    /** Flag whether Remote attribute present. */
+    private boolean isRemote = false;
+
+    /** Flag whether HasChildren attribute present. */
+    private boolean isHasChildren = false;
+
+    /** Flag whether HasNoChildren attribute present. */
+    private boolean isHasNoChildren = false;
+
+    /**
+     * Initializes an instance of {@link ExtensionListInfo} from the server responses. It also parses the RFC5828 supported list attribute to identify
+     * whether the attribute is present or not. NOTE that this class goal is to present whether an attribute exists or not. It does not interpret any
+     * semantics. For example, if server has sent both HasChildren and HasNoChildren, the 2 getter methods, isHasChildrenAttributePresent, and
+     * isHasNoChildrenAttributePresent will all be returning true.
+     *
+     * @param resp a LIST command response
+     * @throws ParsingException for errors parsing the responses
+     */
+    public ExtensionListInfo(@Nonnull final IMAPResponse resp) throws ParsingException {
+        super(resp);
+        for (int i = 0; i < attrs.length; i++) {
+            // case insensitive to be consistent with ListInfo parsing
+            if (NON_EXISTENT.equalsIgnoreCase(attrs[i])) {
+                isNonExistent = true;
```

Comment:
can you also updated canOpen in the parent class?

Reply:
Done.